doc/bash-completion/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+# Detects the best place to install bash completion support
+# - Windows is not supported
+# - TODO: macOS support can be provided through Homebrew
+IF(NOT LMMS_BUILD_WIN32 AND NOT LMMS_BUILD_APPLE)
+
+	# standard install location for most distributions
+	SET(BASHCOMP_FALLBACK "/usr/share/bash-completion/completions")
+
+	# check for common portable installation (CMAKE_INSTALL_PREFIX=../target)
+	GET_FILENAME_COMPONENT(BINARY_ADJ_INSTALL "${CMAKE_BINARY_DIR}/../target" ABSOLUTE)
+	IF("${BINARY_ADJ_INSTALL}" STREQUAL "${CMAKE_INSTALL_PREFIX}")
+		SET(SKIP_PKG_CHECK TRUE)
+	ENDIF()
+
+	# check for other signs of user-level install
+	SET(USER_LEVEL_INSTALL "/home/;/root/;/Users/")
+	FOREACH(_startswith ${USER_LEVEL_INSTALL})
+		IF("${CMAKE_INSTALL_PREFIX}" MATCHES "^${_startswith}")
+			SET(SKIP_PKG_CHECK TRUE)
+			BREAK()
+		ENDIF()
+	ENDFOREACH()
+
+	# determine BASHCOMP_PKG_PATH variable
+	IF(SKIP_PKG_CHECK)
+		SET(BASHCOMP_PKG_PATH "${CMAKE_INSTALL_PREFIX}${BASHCOMP_FALLBACK}")
+	ELSE()
+		PKG_CHECK_MODULES(BASH_COMPLETION bash-completion)
+		PKG_GET_VARIABLE(BASHCOMP_PKG_PATH bash-completion completionsdir)
+	ENDIF()
+
+	# final fallback for non-user-level installs
+	IF("${BASHCOMP_PKG_PATH}" STREQUAL "")
```

Comment:
If you determine whether to install the completion script in this way, it will always be installed if `BASHCOMP_PKG_PATH` is defined. For example, `cmake . -DCMAKE_INSTALL_PREFIX=/tmp` gives:
```
-- Bash completions will be installed to /usr/share/bash-completion/completions
```

Reply:
Yeah I discussed this with @JohannesLorenz.  I hadn't considered `/tmp`, we can add that to the user-space whitelist.  There's not too many places that are utilized like this and we forgot about `/tmp`.  Are you ok with simply adding it?

Some may argue that any mountpoint can be valid, but I'm not sure how far down this rabbit hole you want to go.  In the end, I'd like both AppImages and Homebrew builds to be able to get to bash completion, even if they won't be used for some time.

Reply:
I wasn't talking about a specific path. There are some more spaces that user can write, e.g. `/var/tmp`, some subdirectories of `/media`/`/mnt`(some mount points), but I'm not sure how should we handle them.

Reply:
Agreed.  @JohannesLorenz' first proposal was to check to see if the directory was writable.  Perhaps that's better overall?

Reply:
> The install paths leading to an install of the completion should be a white-list, e.g. only /usr and /usr/share.

This whitelist is useless if we can't detect a non-root install.

Reply:
Checking for writable requires the ugly shell script.

@tresf  Why do we need to detect a non-root install? What's wrong with my last commit from yesterday?

Reply:
> @tresf Why do we need to detect a non-root install? What's wrong with my last commit from yesterday?

Because I want the completions part of the AppImages and eventually available through Homebrew, neither needing `sudo/root`.

Reply:
For AppImages, I think we wanted to use `../target`. For Homebrew, no idea what you are talking about. Probably best if you make another commit we can check...